### DISCUSSION TITLE
Is vue-tsc a full replacement of tsc?

### DISCUSSION THREAD

**Discussion by jods4** [johnsoncodehk/vue-tsc#48] [Q&A]
The README mentions doing a type-check only with `--noEmit` and then using `vite` for regular build.
Is `vue-tsc` a full, compatible, typescript compiler though?

Can I build a basic project on the CLI using `vue-tsc` without the "no emit" flag?

More interestingly: can I use it as a custom TSC compiler in loaders such as `rollup-plugin-typescript2`?

Some background / context:
- I have a Vite build that is customized to use the offical tsc compiler instead of esbuild (main reason being const enum support).
- The build uses `rollup-plugin-typescript2` to achieve that, which has an option to use a specific compiler.
- The awesome Volar types can't be injected into regular `tsc`, so I'm left with 2 options if I want to typecheck my project:
- (1) do 2 passes, `vue-tsc --noEmit && vite build`, which would in my case go through TS compiler twice, so that's a bit of a bummer speedwise, but it'd work;
- (2) swap `vue-tsc` for `typescript` in `rollup-plugin-typescript2`, hoping it would compile and do the full typechecking magic.

I suppose (2) would require some fiddling with the `vue` rollup plugin, which compiles SFC files, so that each plugin sees the correct input. Otherwise `rollup-plugin-typescript2` sees already compiled `.vue` files and that would ruin the `vue-tsc` magic.

**Comment by johnsoncodehk** [comment]
Sorry I don't know how `rollup-plugin-typescript2` working for now, I will give you answer in future.
